### PR TITLE
314 display real stats current ones are mocks

### PR DESCRIPTION
**Pull request by Apoorva64**

### CODE REVIEW

**Comment by Apoorva64** [comment]
# ☁️ Nx Cloud Report

CI is running/has finished running commands for commit 2178a20026ca1875f0ec47a22ceca41f0e3490f5. As they complete they will appear below. Click to see the status, the terminal output, and the build insights.

[📂 See all runs for this branch](https://cloud.nx.app/branch?workspaceId=640ee8fb56da4d6500e89717&branchName=315,314-display-real-stats-current-ones-are-mocks)

---


<details>
<summary>✅ Successfully ran 1 target </summary>

- [`nx serve backend-api`](https://cloud.nx.app/runs/ImfqKk1XfZ)
</details>

---

Sent with 💌 from [NxCloud.](https://nx.app)

**Review by Guekka — APPROVED** [review]
Did not read the code yet, but approving in case you want to merge

File: apps/backend/api/src/routes/question-result.route.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    Schema.QuestionResultWhereUniqueInputSchema,
+    undefined,
+    undefined
+  ),
+  asyncHandler(deleteQuestionResultHandler)
+);
+
+questionResultRouter.get(
+  '/',
+  validateSplit(undefined, Schema.QuestionResultWhereInputSchema, undefined),
```
Comment (Guekka) [inline apps/backend/api/src/routes/question-result.route.ts:58]: I don't think we need a `QuestionResultWhereInputSchema` for this one?

File: apps/front-end/front-office/src/app/reducers/game/game.effects.ts [hunk 2]
```diff
@@ -55,12 +60,30 @@ export class GameEffects {
...
+                        id: quiz.data.id,
+                      },
+                    },
+                    patient: {
+                      connect: {
+                        id: patient.id,
+                      },
+                    },
+                  })
+                  .pipe(
```
Comment (Guekka) [inline apps/front-end/front-office/src/app/reducers/game/game.effects.ts:77]: No need to `pipe(x).pipe(y)`, you can `pipe(x, y)`

File: apps/front-end/front-office/src/app/reducers/game/game.effects.ts [hunk 3]
```diff
@@ -73,15 +96,39 @@ export class GameEffects {
...
+                id: state.currentQuestion.id,
+              },
+            },
+            quizSession: {
+              connect: {
+                id: state.quizSession.id,
+              },
+            },
+          })
+          .pipe(
```
Comment (Guekka) [inline apps/front-end/front-office/src/app/reducers/game/game.effects.ts:127]: Same here, no need for two `pipe`

**Comment by Guekka** [comment]
Just tried it, it does not compile. I assume you already know about that

File: apps/front-end/front-office/src/app/reducers/game/game.effects.ts (hunk 2 above)
Comment (Apoorva64) [reply apps/front-end/front-office/src/app/reducers/game/game.effects.ts:77]: there's a mergeMap

**Review by Guekka — CHANGES_REQUESTED** [review]
Resulting graph looks weird
image

Also, "pourcentage de réussite" doesn't seem to work

**Review by Guekka — CHANGES_REQUESTED** [review]
"Pourcentage de réussite" is still wrong. Maybe you have to multiply it by 100?

**Review by Guekka — DISMISSED** [review]

**Review by Guekka — APPROVED** [review]
LGTM :+1:

**Pull request merged**